enhancements/cguConditionEnhancement.md

```diff
@@ -0,0 +1,452 @@
+  }
+]
+```
+
+- **Network operator** has below condition:
+
+```yaml
+[root@hv2 ~]# oc get co/network -ojson | jq '.status.conditions'
+[
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "False",
+    "type": "Degraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "False",
+    "type": "ManagementStateDegraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "True",
+    "type": "Upgradeable"
+  },
+  {
+    "lastTransitionTime": "2022-06-07T15:35:25Z",
+    "status": "False",
+    "type": "Progressing"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:37:57Z",
+    "status": "True",
+    "type": "Available"
+  }
+]
+```
+
+## Proposal
+
+This section proposes minimalistic changes without breaking the current design pattern. The below proposal/recommendation consists of more granular `status.conditions`  based on all the verification we already have in TALM.
+
+**Considering**: backup,precaching is true and enable for upgrade is true as well. 
+
+### Workflow Description
+
+1. Condition type related to cluster existing in the managedcluster. The type could reflect on the existence of the targeted cluster to be present in the managedcluster. 
+
+  ```yaml
+  Type:    "ClusterValidation/ClustersSelected",
+  Status:  True/False,
+  Reason:  "",
+  Message: “”
+  ```
+
+2. Condition type: **BackupDone**
+3. a) Spec validation for the `managed policies`. Currently it gets checked both during pre-caching and for upgrade. This check can be done before starting an upgrade, irrespective of pre-caching being true or false. b) Platform related verification. 
+Point a) and b) can be merged into PreUpgradeValidation where we check managed policies existence, conformity of platform image etc.
+
+ ```yaml
+  Type:    "PreUpgradeValidation",
+  Status:  True/False,
+  Reason:  "MissingManagedPolicy/InvalidPlatformImage/ValidationComplete",
+  Message: “”
+  ```
+
+4. Condition type: **PrecacheSpecValid** and **PrecachingDone** which is already in the current design. For **PrecacheSpecValid**, we may omit the highlighted Reason from the [current spec](#precachingSpec)
+5. Checking the blocking CRs status, if there is any issue UpgradeCannotStart reason is reported. If CRs are not blocked the upgrade can start. Also, adding and deleting labels to the cluster can be performed here. Condition type: Started/Progressing?
+
+ ```yaml
+    Type:    "Started/Progressing",
+    Status:  True/False,
+    Reason:  "MissingBlockingCR/IncompleteBlockingCR/MissingClusterLabel/InProgress",
+    Message: “”
+  ```
+
+  > Mohamed had a comment on this block: "In my opinion those are 2 different conditions 1) Initialized/Started means that the action (copying policies) is happening  2) Progressing/InProgress means that policy remediation started." To discuss.
+
+  6. The **UpgradeCompleted** and **UpgradeTimedOut** Reason can be moved to condition type: Available/Ready
```

Comment:
The "Available/Ready" is a bit misleading to me since TALM isn't making things available. Maybe "UpdateCompleted"?

Reply:
We discussed it in the scrum meeting with @jc-rh  and @serngawy. We have found Completed/succeeded can be a better alternative. wdyt?

Reply:
Sounds good to me.